Client/Frontend/Library/Bookmarks/BookmarksPanel.swift

```diff
@@ -15,56 +15,24 @@ let LocalizedRootBookmarkFolderStrings = [
     LocalDesktopFolder.localDesktopFolderGuid: String.Bookmarks.Menu.DesktopBookmarks
 ]
 
-private class SeparatorTableViewCell: OneLineTableViewCell {
-    override func applyTheme() {
-        super.applyTheme()
-
-        backgroundColor = UIColor.theme.tableView.headerBackground
-    }
-}
-
 class BookmarksPanel: SiteTableViewController, LibraryPanel, CanRemoveQuickActionBookmark, Loggable {
 
-    private struct UX {
+    struct UX {
         static let FolderIconSize = CGSize(width: 24, height: 24)
         static let RowFlashDelay: TimeInterval = 0.4
     }
 
-    private enum BookmarksSection: Int, CaseIterable {
-        case bookmarks
-    }
-
     // MARK: - Properties
     var libraryPanelDelegate: LibraryPanelDelegate?
     var notificationCenter: NotificationCenter
     var state: LibraryPanelMainState
-
-    private let bookmarkFolderGUID: GUID
-    private var bookmarkFolder: BookmarkFolderData?
-    private var bookmarkNodes = [FxBookmarkNode]()
-    private var chevronImage = UIImage(named: ImageIdentifiers.menuChevron)
-    private var flashLastRowOnNextReload = false
-
-    private var isRootNode: Bool {
-        return bookmarkFolderGUID == BookmarkRoots.MobileFolderGUID
-    }
-
-    private lazy var bookmarkFolderIconNormal = {
-        return UIImage(named: ImageIdentifiers.bookmarkFolder)?
-            .createScaled(UX.FolderIconSize)
-            .tinted(withColor: UIColor.Photon.Grey90)
-    }()
-
-    private lazy var bookmarkFolderIconDark = {
-        return UIImage(named: ImageIdentifiers.bookmarkFolder)?
-            .createScaled(UX.FolderIconSize)
-            .tinted(withColor: UIColor.Photon.Grey10)
-    }()
+    let viewModel: BookmarksPanelViewModel
 
     // MARK: - Toolbar items
     var bottomToolbarItems: [UIBarButtonItem] {
-        // Return empty toolbar when bookmarks is in root node
-        guard case .bookmarks = state else { return [UIBarButtonItem]() }
+        // Return empty toolbar when bookmarks is in desktop folder node
+        guard case .bookmarks = state,
+              viewModel.bookmarkFolderGUID != LocalDesktopFolder.localDesktopFolderGuid else { return [UIBarButtonItem]() }
```

Comment:
nit: `else` should be on its own line, closing off the `guard` block visually nicely